### PR TITLE
[core] [react] Remove "strictNullChecks: false" tsconfig property

### PR DESCRIPTION
**Pull request by sc-nikolaoslazaridis**
- [x] This PR follows the [Contribution Guide](https://github.com/Sitecore/content-sdk/blob/dev/CONTRIBUTING.md)
- [x] Changelog updated
- [ ] Upgrade guide entry added

## Description / Motivation
- Removed the strictNullChecks option from the tsconfig
- Fixed all errors related to the removed option

## Testing Details

- [x] Unit Test Added
- [ ] Manual Test/Other (Please elaborate)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/core/src/layout/utils.test.ts [hunk 1]
```diff
@@ -68,6 +68,20 @@ describe('core layout utils', () => {
...
+      expect(result).to.be.true;
+    });
+
+    it('should return true if field is null', () => {
+      const field = null;
+      const result = isFieldValueEmpty(field);
+
+      expect(result).to.be.true;
+    });
+
```
Comment (illiakovalenko) [inline packages/core/src/layout/utils.test.ts]: We already have unit tests covering these scenarios, please, double check

File: packages/react/src/components/DesignLibrary/DesignLibrary.tsx [hunk 2]
```diff
@@ -135,11 +139,13 @@ export const DesignLibrary = withLoadImportMap(({ loadImportMap }: DesignLibrary
         setRenderKey((k) => k + 1);
       });
 
-      const importMapEvent = getDesignLibraryImportMapEvent(rendering.uid, importMap);
+      const importMapEvent = getDesignLibraryImportMapEvent(uid, importMap);
       postToDesignLibrary(importMapEvent);
 
+      if (!propsState.fields || !propsState.params) return;
```
Comment (illiakovalenko) [inline packages/react/src/components/DesignLibrary/DesignLibrary.tsx]: Let's don't skip _componentProps_ even when fields or params are undefined. Instead, in [getDesignLibraryComponentPropsEvent](https://github.com/Sitecore/content-sdk/blob/chore/JSS-8592/packages/core/src/editing/codegen/preview.ts#L396-L397) you can make `fields` and `params` optional and set default parameters value as `{}`

File: packages/react/src/components/DesignLibrary/DesignLibraryClientEvents.tsx [hunk 3]
```diff
@@ -70,31 +73,33 @@ export const DesignLibraryVariantGenerationEvents = ({
...
-      const importMapEvent = getDesignLibraryImportMapEvent(component.uid, importMap);
+      const importMapEvent = getDesignLibraryImportMapEvent(component.uid, importMap!);
       postToDesignLibrary(importMapEvent);
 
       const propsEvent = getDesignLibraryComponentPropsEvent(
         component.uid,
-        component.fields,
-        component.params
+        component.fields || {},
+        component.params || {}
```
Comment (illiakovalenko) [inline packages/react/src/components/DesignLibrary/DesignLibraryClientEvents.tsx]: See my comment regarding `getDesignLibraryComponentPropsEvent` function, we can make these parameters optional there and apply default value

File: packages/react/src/components/ErrorBoundary.tsx [hunk 4]
```diff
@@ -69,7 +78,7 @@ class ErrorBoundary extends React.Component<ErrorBoundaryProps> {
         } else {
```
Comment (illiakovalenko) [inline packages/react/src/components/ErrorBoundary.tsx:84]: Can we reuse `ErrorComponent` in "if" branch as well?

File: packages/react/src/components/ErrorBoundary.tsx [hunk 5]
```diff
@@ -18,14 +18,23 @@ export type ErrorBoundaryProps = {
...
 
+/**
+ * Simple error component applying basic error styling.
+ * @param {string} message The error message to display.
+ * @returns A JSX element containing the error message in a div.
+ * @internal
+ */
+export const ErrorComponent = ({ message }: { message: string }) => {
+  return <div className="sc-content-sdk-placeholder-error">{message}</div>;
+};
```
Comment (illiakovalenko) [inline packages/react/src/components/ErrorBoundary.tsx]: Let's add a simple unit test to cover that it renders as expected

File: packages/react/src/components/FEaaS/BYOCWrapper.tsx [hunk 6]
```diff
@@ -59,11 +59,9 @@ export class BYOCComponent extends React.Component<BYOCComponentProps> {
...
         errorOverride: 'BYOC: The ComponentName for this rendering is missing',
       };
-      return props.missingComponentComponent ? (
-        <this.props.missingComponentComponent {...noNameProps} />
-      ) : (
-        <MissingComponent {...noNameProps} />
-      );
+      const MissingComp = this.props.missingComponentComponent;
+
+      return MissingComp ? <MissingComp {...noNameProps} /> : <MissingComponent {...noNameProps} />;
```
Comment (illiakovalenko) [inline packages/react/src/components/FEaaS/BYOCWrapper.tsx:65]: Do we need to apply these changes? I'm asking because in the same component we have two extra similar cases

File: packages/react/src/components/FEaaS/FEaaSWrapper.tsx [hunk 7]
```diff
@@ -35,10 +35,10 @@ export const FEaaSComponent = (props: FEaaSComponentProps): JSX.Element => {
...
       data={data}
       template={props.template}
-      cdn={props.params?.ComponentHostName}
-      library={props.params?.LibraryId}
-      version={props.params?.ComponentVersion}
-      component={props.params?.ComponentId}
+      cdn={props.params?.ComponentHostName as string}
+      library={props.params?.LibraryId as string}
+      version={props.params?.ComponentVersion as string}
+      component={props.params?.ComponentId as string}
```
Comment (illiakovalenko) [inline packages/react/src/components/FEaaS/FEaaSWrapper.tsx]: Type cast is not really needed here, in case of `library` and `component` we can just fallback to empty string. Also update the existing [unit test](https://github.com/Sitecore/content-sdk/blob/45c3ea7cdadb778e1bf283801dc56895670158d5/packages/react/src/components/FEaaS/FEaaSWrapper.test.tsx#L60)
```suggestion
      cdn={props.params?.ComponentHostName}
      library={props.params?.LibraryId ?? ''}
      version={props.params?.ComponentVersion}
      component={props.params?.ComponentId ?? ''}
```

File: packages/react/src/components/FEaaS/feaas-utils.ts [hunk 8]
```diff
@@ -110,12 +122,13 @@ async function fetchData(dataOptions: FEAAS.DataOptions): Promise<FEAAS.DataScop
...
  */
 export const composeComponentEndpoint = (
-  params: FEaaSComponentParams,
+  params: FEaaSComponentParams & RequiredFEaaSParams,
   revisionFallback: RevisionType
 ) => {
   const revision = params.ComponentRevision || revisionFallback;
   const hostname = params.ComponentHostName.startsWith('https://')
     ? params.ComponentHostName
     : `https://${params.ComponentHostName}`;
```
Comment (illiakovalenko) [inline packages/react/src/components/FEaaS/feaas-utils.ts]: For simplicity you can remove `RequiredFEaaSParams`, `isFEaaSComponentParamsComplete`, `ValidatedFEaaSComponentParams` things and just add optional chaining for `ComponentHostName`
```suggestion
  params: FEaaSComponentParams,
  revisionFallback: RevisionType
) => {
  const revision = params.ComponentRevision || revisionFallback;
  const hostname = params.ComponentHostName?.startsWith('https://')
    ? params.ComponentHostName
    : `https://${params.ComponentHostName}`;
```

File: packages/react/src/components/FEaaS/models.ts
Comment (illiakovalenko) [inline packages/react/src/components/FEaaS/models.ts:1]: See my comment in `feaas-utils`, all the types can be reverted for simplicity

File: packages/react/src/components/Placeholder/PlaceholderMetadata.tsx
Comment (illiakovalenko) [inline packages/react/src/components/Placeholder/PlaceholderMetadata.tsx:1]: To simplify all the changes you can just:
* Mark `id` as optional in `getCodeBlockAttributes`
* Mark `id` as optional in `renderComponent`

File: packages/react/src/components/Placeholder/models.ts [hunk 9]
```diff
@@ -21,7 +21,7 @@ export interface AppComponentProps {
     [name: string]: Field | Item | Item[];
   };
   params: {
-    [name: string]: string;
+    [name: string]: string | undefined;
```
Comment (illiakovalenko) [inline packages/react/src/components/Placeholder/models.ts]: Let's don't update the interface, [here](https://github.com/Sitecore/content-sdk/blob/dev/packages/react/src/components/Placeholder/placeholder-utils.tsx#L91) we can return `{ styles: '' }`

File: packages/react/src/enhancers/withAppPlaceholder.test.tsx [hunk 10]
```diff
@@ -96,6 +96,32 @@ describe('withAppPlaceholder HOC', () => {
   });
 
   // Basic functionality tests
+  it('should render without placeholders', () => {
+    // Create simple test data with only valid components
```
Comment (illiakovalenko) [inline packages/react/src/enhancers/withAppPlaceholder.test.tsx]: :)
```suggestion
```

**Review by illiakovalenko — CHANGES_REQUESTED** [review]
Good cleanup! 👍
please take a look at my comments below

File: packages/react/src/components/FEaaS/BYOCWrapper.tsx (hunk 6 above)
Comment (sc-nikolaoslazaridis) [reply packages/react/src/components/FEaaS/BYOCWrapper.tsx:65]: We get an error here if i wont rearrange the code. This is not a functional change.

File: packages/react/src/components/ErrorBoundary.tsx (hunk 4 above)
Comment (sc-nikolaoslazaridis) [reply packages/react/src/components/ErrorBoundary.tsx:84]: I made the ErrorComponent accept either message or children and used it in the "if" also

File: packages/core/src/editing/codegen/preview.ts [hunk 11]
```diff
@@ -393,8 +393,8 @@ export function getDesignLibraryComponentPreviewErrorEvent(
  */
 export function getDesignLibraryComponentPropsEvent(
   uid: string,
-  fields: ComponentFields,
-  parameters: ComponentParams
+  fields: ComponentFields = {},
+  parameters: ComponentParams = {}
```
Comment (illiakovalenko) [inline packages/core/src/editing/codegen/preview.ts:397]: To don't decrease a coverage let's add a simple unit test to test a scenario with empty "fields", "parameters"

File: packages/react/src/components/ErrorBoundary.tsx [hunk 12]
```diff
@@ -18,14 +18,29 @@ export type ErrorBoundaryProps = {
...
 };
 
+/**
+ * Simple error component applying basic error styling.
+ * @param {object} props - Either with `message` (string) or with `children` (ReactNode), but not both.
+ */
+export const ErrorComponent = (
+  props:
+    | { message: React.ReactNode; children?: never }
+    | { children: React.ReactNode; message?: never }
```
Comment (illiakovalenko) [inline packages/react/src/components/ErrorBoundary.tsx:28]:
```suggestion
  props: { children?: React.ReactNode; message?: React.ReactNode; }
```

**Review by illiakovalenko — APPROVED** [review]
Approved in advance 👍
Please, look and resolve a few comments below then feel free to merge

File: packages/react/src/components/ErrorBoundary.tsx (hunk 12 above)
Comment (sc-nikolaoslazaridis) [reply packages/react/src/components/ErrorBoundary.tsx:28]: I want to enforce it on typescript level, either use one or the other
